packages/ic-management/src/ic-management.canister.ts

```diff
@@ -350,4 +355,41 @@ export class ICManagementCanister {
       canister_id: canisterId,
     });
   };
+
+  /**
+   * This method takes a snapshot of the specified canister. A snapshot consists of the wasm memory, stable memory, certified variables, wasm chunk store and wasm binary.
+   *
+   * @link https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-take_canister_snapshot
+   *
+   * @param {Object} params - Parameters for the snapshot operation.
+   * @param {Principal} params.canisterId - The ID of the canister for which the snapshot will be taken.
+   * @param {SnapshotIdText | snapshot_id} [params.snapshotId] - The ID of the snapshot to replace, if applicable.
+   * Can be provided as a `string` or a `Uint8Array`.
+   * If not provided, a new snapshot will be created.
+   *
+   * @returns {Promise<take_canister_snapshot_result>} A promise that resolves with the snapshot details,
+   * including the snapshot ID, total size, and timestamp.
+   *
+   * @throws {Error} If the snapshot operation fails.
+   */
+  takeCanisterSnapshot = ({
+    canisterId,
+    snapshotId,
+  }: {
+    canisterId: Principal;
+    snapshotId?: SnapshotIdText | snapshot_id;
+  }): Promise<take_canister_snapshot_result> => {
+    const { take_canister_snapshot } = this.service;
+
+    return take_canister_snapshot({
+      canister_id: canisterId,
+      replace_snapshot: toNullable(
+        nonNullish(snapshotId)
```

Comment:
FUTURE development: maybe this part of parsing `SnapshotIdText | snapshot_id` can go in another small util (for another PR obviously), like something to parse `SnapshotIdText | snapshot_id` to the correct type always, for example `snapshot_id` only

Reply:
Good idea. There is 2-3 other functions to add too, let's see if there is also some things that should be reused between those.